common/thorhelper/thorcommon.cpp

```diff
@@ -2023,14 +2023,28 @@ static IOutputMetaData *_getDaliLayoutInfo(MemoryBuffer &layoutBin, IPropertyTre
                 error.setown(E); // Save to throw later if we can't recover via ECL
             }
         }
-        if (props.hasProp("ECL"))
+        if (props.hasProp("meta"))
+        {
+            props.getPropBin("meta", layoutBin);
+            try
+            {
+                return createTypeInfoOutputMetaData(layoutBin, isGrouped);
+            }
+            catch (IException *E)
+            {
+                EXCLOG(E);
+                error.setown(E); // Save to throw later if we can't recover via ECL
+            }
+        }
+        const char * layoutECL = props.queryProp("ECL");
+        if (!layoutECL)
+            layoutECL = props.queryProp("@ecl");
```

Comment:
Why? The ECL is always multi-line - not sure it can ever be stored in an XML attribute (though maybe can in yaml?)

Reply:
Yes, this is for the yaml definition.  You can have a newline in an attribute it will be quoted as &#10;